src/app/lifecycle.ts

```diff
@@ -30,48 +31,97 @@ export class AppStore {
   private readonly appStore = new Map<string, FirebaseApp>();
 
   public initializeApp(options?: AppOptions, appName: string = DEFAULT_APP_NAME): App {
+    validateAppNameFormat(appName);
+
+    let autoInit = false;
     if (typeof options === 'undefined') {
+      autoInit = true
       options = loadOptionsFromEnvVar();
       options.credential = getApplicationDefault();
     }
 
-    if (typeof appName !== 'string' || appName === '') {
-      throw new FirebaseAppError(
-        AppErrorCodes.INVALID_APP_NAME,
-        `Invalid Firebase app name "${appName}" provided. App name must be a non-empty string.`,
-      );
-    } else if (this.appStore.has(appName)) {
-      if (appName === DEFAULT_APP_NAME) {
+    // Check if an App already exists and, if so, ensure its AppOptions match
+    // those of this `initializeApp` request. 
+    if (this.appStore.has(appName)) {
+      const currentApp = this.appStore.get(appName)!;
+      // Ensure the autoInit state matches the existing app's. If not, throw.
+      if (currentApp.autoInit() !== autoInit) {
         throw new FirebaseAppError(
-          AppErrorCodes.DUPLICATE_APP,
-          'The default Firebase app already exists. This means you called initializeApp() ' +
-          'more than once without providing an app name as the second argument. In most cases ' +
-          'you only need to call initializeApp() once. But if you do want to initialize ' +
-          'multiple apps, pass a second argument to initializeApp() to give each app a unique ' +
-          'name.',
-        );
+          AppErrorCodes.INVALID_APP_OPTIONS,
+          `Firebase app named "${appName}" attempted mismatch between custom AppOptions` +
```

Comment:
Can we simplify this error message? I am not sure `Auto Init` is a well known term in the Admin SDK world